services/utils/__init__.py

```diff
@@ -247,19 +255,44 @@ def _is_valid_dsn(dsn):
 
     @property
     def connection_string_url(self):
-        # postgresql://[user[:password]@][host][:port][/dbname][?param1=value1&...]
-        return f'postgresql://{quote(self._user)}:{quote(self._password)}@{self._host}:{self._port}/{self._database_name}?sslmode=disable'
+        base_url = f'postgresql://{quote(self._user)}:{quote(self._password)}@{self._host}:{self._port}/{self._database_name}'
+        if (self._ssl_mode in ['allow', 'prefer', 'require', 'verify-ca', 'verify-full']):
+            ssl_query = f'sslmode={self._ssl_mode}'
+            if self._ssl_cert_path is not None:
+                ssl_query = f'{ssl_query}&sslcert={self._ssl_cert_path}'
+            if self._ssl_key_path is not None:
+                ssl_query = f'{ssl_query}&sslkey={self._ssl_key_path}'
+            if self._ssl_root_cert_path is not None:
+                ssl_query = f'{ssl_query}&sslrootcert={self._ssl_root_cert_path}'
+        else:
+            ssl_query = f'sslmode=disable'
+
+        return f'{base_url}?{ssl_query}'
 
     @property
     def dsn(self):
         if self._dsn is None:
-            return psycopg2.extensions.make_dsn(
-                dbname=self._database_name,
-                user=self._user,
-                host=self._host,
-                port=self._port,
-                password=self._password
-            )
+            ssl_mode = self._ssl_mode
+            sslcert = self._ssl_cert_path
+            sslkey = self._ssl_key_path
+            sslrootcert = self._ssl_root_cert_path
+            if (ssl_mode not in ['allow', 'prefer', 'require', 'verify-ca', 'verify-full']):
+                ssl_mode = 'disable'
```

Comment:
Tests are currently not passing due to some assertions for the integration tests being made, and this change is appending `sslmode=disable` to the DSN. A quick fix would be to instead set `ssl_mode=None` here to keep the existing functionality intact.

I'm also wondering whether explicitly passing `sslmode=disable` will be a subtle change in behavior, as I checked and the default value for the sslmode in postgres seems to be `prefer` instead of `disable`.